src/wallet/wallet_list.h

```diff
@@ -41,4 +41,33 @@
  */
 uint8_t get_wallet_list(const char *wallet_list[]);
 
+/**
+ * @brief This API searches for wallet on the flash using wallet_id as key
+ * @details If a VALID_WALLET is found in the flash, then this API fills the
+ * wallet_name, wallet_info, wallet_id fields of the Wallet structure
+ *
+ * @param wallet_id The wallet_id that needs to be searched
+ * @param wallet Reference to Wallet structure which will be populated by the
+ * function
+ * @return true If the wallet corresponding to wallet_id is found in the flash
+ * and is in VALID_WALLET state.
+ * @return false If the wallet corresponding to wallet_id is not found or is not
+ * in VALID_WALLET state.
+ */
+bool wallet_selector(const uint8_t *wallet_id, Wallet *wallet);
```

Comment:
`get_valid_wallet_data` seems like a better name for this function as it actually fetches relevant data for a valid wallet.

Reply:
Also brief comment is same for both APIs here, which is very confusing

Reply:
Updated name to `get_wallet_data_by_id`